yt_dlp/extractor/fancode.py

```diff
@@ -74,8 +107,12 @@ def _real_extract(self, url):
             raise ExtractorError('Unable to extract brightcove Video ID')
 
         is_premium = media.get('isPremium')
-        if is_premium:
-            self.report_warning('this video requires a premium account', video_id)
+        is_available = media.get('isUserEntitled')
+
+        if is_premium and self._ACCESS_TOKEN is None:
+            self.raise_login_required()
```

Comment:
Do cookies work for login? If not, use:
```suggestion
            self.raise_login_required(method='password')
```
Same everywhere else

Reply:
Actually, don't do this. Its better to modify the `raise_login_required` function to allow a custom message as `method` and tell the user to use the `-u "refresh" -p <refresh_token>`